src/components/InterestForm/index.tsx

```diff
@@ -0,0 +1,83 @@
+'use client';
+
+import { useState } from 'react';
+import { UUID } from 'crypto';
+import { insertInterest } from '../../api/supabase/queries/interest';
+import { Interest, CaseListing } from '../../types/schema';
+import { Button } from '../Button';
+import {
+  FormContainer,
+  FormTextArea,
+  FormInput,
+  RadioGroup,
+  RadioLabel,
+  RadioInput,
+} from './styles';
+
+const radioOptions = ['Attorney', 'Interpreter', 'Both Attorney & Interpreter'];
+
+export default function InterestForm({ caseData }: { caseData: CaseListing }) {
+  const [reason, setReason] = useState<string>('');
+  const [rolesInterested, setRolesInterested] = useState<string>('');
+  const [startDate, setStartDate] = useState<string>('');
+
+  const handleInsert = async () => {
```

Comment:
Curious why this function seems to force a re-render of the whole page? Are we explicitly doing this somewhere?

Reply:
I have a rough idea of why it might be re-rendering, but I'll also look more into this during the next sprint! I think once I reconnect the form to the backend it'll be easier to debug.